### ISSUE TITLE
npm package missing .d.ts files

### ISSUE THREAD

**Issue by veteran29** [winseros/gulp-armapbo-plugin#2]
Despite having `declaration: true` in tsconfig.json, package published to npm does not have module declaration files, these files should be published to allow working with this plugin with typescript.

http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

BTW: Thanks for great plugin/library.

**Comment by winseros** [comment]
Hi, may I ask why do you need `*d.ts` for a gulp plugin?

**Comment by veteran29** [comment]
I have written gulpfile in TypeScript.

If you have ts-node installed gulp can run `gulpfile.ts` and there is no need to compile it.

For example TypeScript is compiled with gulp and has [gulpfile.ts](https://github.com/Microsoft/TypeScript/blob/master/Gulpfile.ts).

**Comment by winseros** [comment]
Ok, then I'll add type definitions shortly.

**Comment by winseros** [comment]
Please take a look at #3. If it is acceptable for you, then I merge the PR. Please note I had to change plugin export signature severely to be able to additionally export TypeScript interfaces.

#### Before:
``` TypeScript
export = (options?: StreamOptions): PboTransformStream => {
    return new PboTransformStream(options);
};
```
see [main.ts](https://github.com/winseros/gulp-armapbo-plugin/blob/5a375bedd380b8fbbd77ec8d5d7e00c902acb95b/src/main.ts) at master

#### After:
``` TypeScript
export {pack, StreamOptions};
```
see [main.ts](https://github.com/winseros/gulp-armapbo-plugin/blob/21f3ff1b4fef2af7048175adf18f5789133c37d1/src/main.ts) at making-plugin-ts-compatible

Apparently that breaks plugin api backwards compatibility and would lead to the major version increment.

**Comment by veteran29** [comment]
Looks (and works) fine for me. 

Thanks!

**Comment by winseros** [comment]
I merged #3 and published a new major version. Please let me know if everything is ok and I'll close the issue then.

**Comment by veteran29** [comment]
Everything is ok. Closing issue.
